frontend/online-shop/src/pages/products-collection/products-collection.page.tsx

```diff
@@ -0,0 +1,34 @@
+import { FC, useEffect, useState } from "react";
+import "./products-collection.style.css";
+import ProductPreview from "../../components/product-preview/product-preview.component";
+import { Grid } from "@mui/material";
+import { fetchProducts } from "../../services/products.service";
+const ProductsCollection: FC = () => {
+  // eslint-disable-next-line
+  const [product, setProduct] = useState([
+    {
+      id: 0,
+      name: "",
+      description: "",
+      category: "",
+      price: 0,
+      image: "",
+    },
+  ]);
+
+  useEffect(() => {
+    fetchProducts({ setProduct });
```

Comment:
Here you should not pass any parameter to the `fetchProducts`. You should catch the restul here and set the state also here